Kinvey/Kinvey/HttpRequest.swift

```diff
@@ -358,6 +358,16 @@ internal class HttpRequest: TaskProgressRequest, Request {
         if let clientAppVersion = client.clientAppVersion {
             request.setValue(clientAppVersion, forHTTPHeaderField: KinveyHeaderField.clientAppVersion)
         }
+        if let url = request.url,
+            let query = url.query,
+            query.contains("+"),
```

Comment:
those are already handled by the Apple's class `URLQueryItem`, the only one missing is `+` which Apple is being strict to the RFC spec, but W3C recommends to encode. Please look the note section in the link below:

https://developer.apple.com/documentation/foundation/nsurlcomponents/1407752-queryitems